framework/web/UrlManager.php

```diff
@@ -428,6 +429,27 @@ public function createUrl($params)
     }
 
     /**
+     * Returns the value indicating whether result of [[createUrl()]] of rule should be cached in internal cache.
+     *
+     * @param UrlRuleInterface $rule
+     * @return bool|null `true` if result should be cached, `false` if not, `null` if rule does not provide info about
+     * create URL status.
+     * @since 2.0.12
+     * @see getUrlFromCache()
+     * @see setRuleToCache()
+     * @see UrlRule::$createStatus
+     */
+    protected function canBeCached(UrlRuleInterface $rule)
+    {
+        if (isset($rule->createStatus)) {
+            return in_array($rule->createStatus, [
+                UrlRule::CREATE_STATUS_PARAMS_MISMATCH,
```

Comment:
If we want keep cache logic in `UrlManager` I have no idea how to solve this differently.